keps/369-job-interface/README.md

```diff
@@ -0,0 +1,381 @@
+like MPIJob, who lacks the capacity of queueing.
+
+The good news is Kueue is extensible and simple to integrate with through the intermediate medium, we named _Workload_ in Kueue,
+what we need to do is to build a controller to reconcile the workload and the job itself.
+
+But the complexity lies in developers who are familiar with job-like applications may have little knowledge of the
+implementation details of Kueue and they have no idea where to start to build the controller. In this case, if we can provide an
+interface which defines the default behaviors of the Job, and serve the Kubernetes Job as a standard template, it will do them
+a great favor.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Define an interface which shapes the default behaviors of Job
+- Make Kubernetes Job an implementation template of the interface
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Integrate with other job-like applications
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+We collected feedbacks from the community about how to fully integrate Kueue with MPIJob,
+see [#499](https://github.com/kubernetes-sigs/kueue/issues/499).
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Job interface defined here is a hint for developers to build their own controllers,
+it's not a hard constraint(to implement the interface) but a good reference.
+- This will increase the code complexity by wrapping the original Jobs.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
```

Comment:
Can we also provide interfaces for webhook like https://github.com/kubernetes-sigs/kueue/blob/160021fcd342f0b010e4d0d5e64ea634dba54df2/pkg/controller/workload/job/job_webhook.go?